destalinator.py

```diff
@@ -34,7 +38,14 @@ def __init__(self, slacker, slackbot, activated):
         print("output_debug_to_slack_flag is {}".format(self.output_debug_to_slack_flag))
         self.destalinator_activated = activated
         print("destalinator_activated is {}".format(self.destalinator_activated))
-        self.earliest_archive_date = self.config.earliest_archive_date
+
+        archive_date_string = os.getenv(self.config.earliest_archive_date_env_varname)
```

Comment:
When you say “clean this up quite a bit”, do you mean it would allow removing the `if archive_date_string` conditional? Something like:

```python
archive_date_string = os.getenv(self.config.earliest_archive_date_env_varname, "2000-01-01")
year, month, day = [int(x) for x in archive_date_string.split("-")]
self.earliest_archive_date = datetime.date(year, month, day)
```

If so, you’re right—that‘s definitely cleaner. One issue, though: I think that would change the way the code behaves when the environment variable is set to an empty string. At the moment, setting `EARLIEST_ARCHIVE_DATE` to an empty string is equivalent to not setting it at all, but in the code above, setting it to an empty string would cause the program to fail.

As another option that still removes the conditional and keeps the equivalence between an empty string an an unset environment variable, how about:

```python
archive_date_string = os.getenv(self.config.earliest_archive_date_env_varname) or "2000-01-01"
year, month, day = [int(x) for x in archive_date_string.split("-")]
self.earliest_archive_date = datetime.date(year, month, day)
```

(Possibly still pulling "2000-01-01" out into a constant, just for the sake of giving it a meaningful name.)